### PR TITLE
Change the Structure of UI

### PR DESCRIPTION
**Pull request by KunLW**
I have finish the UI modification. Use Function enum in the presenter package to represent each use case we have. This make our screen builder easy to add/remove the functionality in the screen. I delete all the inheritance of the screenBuilder since we actually can make different screen using the screenBuilder only

### CODE REVIEW

**Review by zeyuzhang516 — DISMISSED** [review]

File: src/main/java/ui/ScreenBuilder.java [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+        buttonController = new CheckProfileController(interactor);
+    }
+
+    public ScreenBuilder(ViewModel dataModel, CheckProfileIGateway gateway) {
+
+        this.dataModel = dataModel;
+        // Add view to the dataModel observable.
+        view = new Integration(dataModel.getFrameName());
+        this.dataModel.addObserver(view);
+        CheckProfilePresenter presenter =new CheckProfilePresenter();
```
Comment (tlan4) [inline src/main/java/ui/ScreenBuilder.java:43]: Shouldn't CheckProfilePresenter take in the dataModel?

**Review by tlan4 — COMMENTED** [review]
With ScreenBuilder we don't need EmployeeBuilder anymore right?

**Review by tlan4 — APPROVED** [review]

**Review by tlan4 — APPROVED** [review]

**Review by staceylry — APPROVED** [review]

**Review by zeyuzhang516 — APPROVED** [review]

File: src/main/java/controller/CheckProfileController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class CheckProfileController implements PropertyChangeListener {
+    private final CheckProfileInputBoundary interactor;
+
+    public CheckProfileController(CheckProfileInputBoundary interactor) {
+        this.interactor = interactor;
+    }
+
+    public void create(Integer requester, Integer target){
+        CheckUserFileRequestModel requestModel = new CheckUserFileRequestModel(requester, target);
+        interactor.checkUserProfile(requestModel);
```
[the remark this answers is not in the document]
Comment (KunLW) [reply src/main/java/controller/CheckProfileController.java:18]: There is no return value, controller just passing the control to the interactor and create then the presenter will create the UI

File: src/main/java/ui/ScreenBuilder.java (hunk 1 above)
Comment (KunLW) [reply src/main/java/ui/ScreenBuilder.java:44]: presenter form the dataModel, which is the ViewModel in MVVP, and that dataModel will control the appearance of UI

**Pull request merged**